src/main/java/randoop/generation/ObjectPool.java

```diff
@@ -0,0 +1,206 @@
+   * not in the pool.
+   *
+   * @param obj the object
+   * @param seq the sequence to be added
+   */
+  @SuppressWarnings("unchecked")
+  /*
+  // Java 9 version
+  public void addOrUpdate(Object obj, Sequence seq) {
+    if (this.objPool.containsKey(obj)) {
+      SimpleList<Sequence> existingSequences = this.objPool.get(obj);
+      this.objPool.put(
+          obj, new ListOfLists<>(existingSequences, new SimpleArrayList<>(List.of(seq))));
+    } else {
+      this.objPool.put(obj, new SimpleArrayList<>(List.of(seq)));
+    }
+  }
+   */
+  public void addOrUpdate(Object obj, Sequence seq) {
+    if (this.objPool.containsKey(obj)) {
+      SimpleList<Sequence> existingSequences = this.objPool.get(obj);
+
+      // Create a new SimpleArrayList and add 'seq' to it
+      SimpleArrayList<Sequence> newList = new SimpleArrayList<>();
+      newList.add(seq);
+
+      // Use ListOfLists with existingSequences and newList
+      this.objPool.put(obj, new ListOfLists<>(existingSequences, newList));
+    } else {
+      // Create a new SimpleArrayList and add 'seq' to it for the else case
+      SimpleArrayList<Sequence> newList = new SimpleArrayList<>();
+      newList.add(seq);
+
+      this.objPool.put(obj, newList);
+    }
+  }
+
+  /**
+   * Filter the sequences in the pool by their type.
+   *
+   * @param t the type to filter by
+   * @return a list of lists of sequences that match the type
+   */
+  public ObjectPool getObjSeqPair(Type t) {
+    ObjectPool objSeqPair = new ObjectPool();
+    for (Object obj : this.objPool.keySet()) {
+      if (equivalentTypes(obj.getClass(), t.getRuntimeClass())) {
+        objSeqPair.put(obj, this.objPool.get(obj));
+      }
+    }
+    return objSeqPair;
+  }
+
+  /**
+   * Get a subset of the object pool that contains objects of a specific type and their sequences.
+   *
+   * @param t the type to filter by
+   * @return a new ObjectPool that contains only the objects of the specified type and their
+   *     sequences
+   */
+  @SuppressWarnings("unchecked")
+  public ListOfLists<Sequence> filterByType(Type t) {
+    ListOfLists<Sequence> filteredSequences = new ListOfLists<>();
+    for (Object obj : this.objPool.keySet()) {
+      if (equivalentTypes(obj.getClass(), t.getRuntimeClass())) {
+        filteredSequences = new ListOfLists<>(filteredSequences, this.objPool.get(obj));
+      }
+    }
+    return filteredSequences;
+  }
+
+  /**
+   * Get a string representation of the object pool.
+   *
+   * @return a string representation of the pool where each line contains an object and its
+   *     associated sequences
+   */
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<Object, SimpleList<Sequence>> entry : this.objPool.entrySet()) {
+      sb.append(entry.getKey().toString())
+          .append(" : ")
+          .append(entry.getValue().toString())
+          .append("\n");
```

Comment:
Don't use "\n" in Java code.  Instead, use `System.lineSeparator()`.

Reply:
Fixed.